crates/uv-distribution/src/workspace.rs

```diff
@@ -824,6 +808,52 @@ fn is_excluded_from_workspace(
     Ok(false)
 }
 
+/// An iterator over a [`Pattern`] in a workspace.
+///
+/// Iterates from the workspace root, returning paths that match the pattern.
+#[derive(Debug)]
+struct WorkspaceGlobIter<'a> {
+    root: &'a Path,
+    glob: &'a Pattern,
+    iter: walkdir::IntoIter,
+}
+
+impl<'a> WorkspaceGlobIter<'a> {
+    fn new(root: &'a Path, glob: &'a Pattern) -> Self {
+        Self {
+            root,
+            glob,
+            iter: walkdir::WalkDir::new(root).into_iter(),
+        }
+    }
+}
+
+impl<'a> Iterator for WorkspaceGlobIter<'a> {
+    type Item = Result<PathBuf, walkdir::Error>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        loop {
+            let entry = match self.iter.next()? {
+                Ok(entry) => entry,
+                Err(err) => return Some(Err(err)),
```

Comment:
What if there's let say a docker `root:root` dir in my project directory (not included in `members`), would we error here?

Reply:
Yeah. For recursive directory traversal, you probably just want to log errors from `Result<DirEntry, walkdir::Error>` and otherwise move on.